packages/admin/admin/src/dataGrid/useDataGridRemote.test.ts

```diff
@@ -1,8 +1,202 @@
+import { GridSortDirection } from "@mui/x-data-grid";
+import { GridFilterModel } from "@mui/x-data-grid/models/gridFilterModel";
+import { renderHook } from "@testing-library/react-hooks";
+import queryString from "query-string";
+import { useHistory, useLocation } from "react-router";
+
+import { useDataGridRemote } from "./useDataGridRemote";
+
+jest.mock("react-router", () => ({
+    ...jest.requireActual("react-router"),
+    useHistory: jest.fn(),
+    useLocation: jest.fn(),
+}));
+
+const mockReplace = () => {
+    const mockedReplace = jest.fn();
+    (useHistory as jest.Mock).mockReturnValue({
```

Comment:
Instead of using `(useHistory as jest.Mock)` I'd do the following:

```tsx
const useHistory = jest.fn();
const useLocation = jest.fn();

jest.mock("react-router", () => ({
    ...jest.requireActual("react-router"),
    useHistory,
    useLocation,
}));

...

useHistory.mockReturnValue(...)
```

Reply:
It works when you move the mocking above the imports: https://stackoverflow.com/questions/65554910/jest-referenceerror-cannot-access-before-initialization

Reply:
Thanks, that works